lib/Backend/FlowGraph.cpp

```diff
@@ -25,27 +147,31 @@ void
 FlowGraph::Build(void)
 {
     Func * func = this->func;
-
     BEGIN_CODEGEN_PHASE(func, Js::FGPeepsPhase);
     this->RunPeeps();
     END_CODEGEN_PHASE(func, Js::FGPeepsPhase);
 
     // We don't optimize fully with SimpleJit. But, when JIT loop body is enabled, we do support
     // bailing out from a simple jitted function to do a full jit of a loop body in the function
-    // (BailOnSimpleJitToFullJitLoopBody). For that purpose, we need the flow from try to catch.
-    if (this->func->HasTry() &&
-        (this->func->DoOptimizeTryCatch() ||
-            (this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody())
-        )
-       )
+    // (BailOnSimpleJitToFullJitLoopBody). For that purpose, we need the flow from try to handler.
+    if (this->func->HasTry() && (this->func->DoOptimizeTryCatch() ||
+            (this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody())))
     {
         this->catchLabelStack = JitAnew(this->alloc, SList<IR::LabelInstr*>, this->alloc);
     }
+    if (this->func->HasFinally() && (this->func->DoOptimizeTryFinally() ||
+        (this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody())))
```

Comment:
>this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody()))) [](start = 9, length = 81)

Take this condition out and check for DoOptimizeTryCatch and DoOptimizeTryFinally inside it. Actually, this makes me think - we can't turn off the opitimizer for try-catch and have it on for try-finally, or vice-versa, right?